synapseclient/models/mixins/table_components.py

```diff
@@ -2295,26 +2661,43 @@ async def main():
         # pandas.read_csv. During implmentation a determination on how large of a CSV
         # that can be loaded from Memory will be needed. When that limit is reached we
         # should continue to force the download of those results to disk.
-
-        # TODO: Replace method in https://sagebionetworks.jira.com/browse/SYNPY-1632
-        results = await loop.run_in_executor(
-            None,
-            lambda: Synapse.get_client(synapse_client=synapse_client).tableQuery(
-                query=query,
-                includeRowIdAndRowVersion=include_row_id_and_row_version,
-                quoteCharacter=quote_character,
-                escapeCharacter=escape_character,
-                lineEnd=line_end,
-                separator=separator,
-                header=header,
-                downloadLocation=download_location,
-            ),
+        result, csv_path = await _table_query(
+            query=query,
+            include_row_id_and_row_version=include_row_id_and_row_version,
+            quote_char=quote_character,
+            escape_char=escape_character,
+            line_end=line_end,
+            separator=separator,
+            header=header,
+            download_location=download_location,
         )
+
         if download_location:
-            return results.filepath
-        return results.asDataFrame(
-            rowIdAndVersionInIndex=False,
-            convert_to_datetime=convert_to_datetime,
+            return csv_path
+
+        date_columns = []
+        list_columns = []
+        dtype = {}
+
+        if result.get("headers") is not None:
+            for column in result.get("headers"):
+                if column["columnType"] == "STRING":
+                    # we want to identify string columns so that pandas doesn't try to
+                    # automatically parse strings in a string column to other data types
+                    dtype[column["name"]] = str
+                elif column["columnType"] in LIST_COLUMN_TYPES:
+                    list_columns.append(column["name"])
+                elif column["columnType"] == "DATE" and convert_to_datetime:
+                    date_columns.append(column["name"])
+
+        return csv_to_pandas_df(
+            filepath=csv_path,
+            separator=kwargs.get("separator", DEFAULT_SEPARATOR),
+            quote_char=kwargs.get("quote_character", DEFAULT_QUOTE_CHARACTER),
+            escape_char=kwargs.get("escape_character", DEFAULT_ESCAPSE_CHAR),
```

Comment:
Since these are already named kw arguments - Do we need to use `kwargs` here? I'd think we could do something like `separator=separator or DEFAULT_SEPARATOR`

Reply:
Since `escape_char` is already defined as an argument to the method call:

```
    @staticmethod
    async def query_async(
        query: str,
        include_row_id_and_row_version: bool = True,
        convert_to_datetime: bool = False,
        download_location=None,
        quote_character='"',
        escape_character="\\",
        line_end=str(os.linesep),
        separator=",",
        header=True,
        *,
        synapse_client: Optional[Synapse] = None,
        **kwargs,
    ) -> Union["DATA_FRAME_TYPE", str]:
```